### PR TITLE
Add the rest of the primitive types

### PR DESCRIPTION
**Pull request by rambleraptor**
This adds the rest of the primitive types in the schema (+ OpenAPI / proto writing)

### CODE REVIEW

File: example/bookstore/v1/bookstore.yaml [hunk 1]
```diff
@@ -8,6 +8,18 @@ resources:
...
+      price:
+        type: FLOAT
+        number: 2
+        required: true
+      published:
+        type: BOOLEAN
+        number: 3
+        required: true
+      edition:
+        type: INT32
```
Comment (toumorokoshi) [inline example/bookstore/v1/bookstore.yaml:20]: I think it's probably better to leave it this way, but as a note there is no int32 type really, and the maximum size 2^53-1 so int64 isn't possible: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER.

I *think* we should probably support it anyway, and maybe implement an int64 as a string in javascript for safety's sake. @hudlow might have thoughts on this.

File: writer/openapi/openapi.go [hunk 2]
```diff
@@ -152,17 +160,21 @@ func convertToOpenAPI(service *parser.ParsedService) *OpenAPI {
...
-	return openAPI
+	return openAPI, nil
 }
 
-func resourceToSchema(r *parser.ParsedResource) Schema {
+func resourceToSchema(r *parser.ParsedResource) (Schema, error) {
 	properties := Properties{}
 	required := []string{}
 	for name, p := range r.Properties {
 		// TODO(YFT): add more handling of types here
```
Comment (toumorokoshi) [inline writer/openapi/openapi.go:170]: remove this comment?

File: writer/proto/resource.go [hunk 3]
```diff
@@ -96,6 +96,18 @@ func GeneratedResourceMessage(r *parser.ParsedResource) (*builder.MessageBuilder
...
+		case schema.Type_INT64:
+			typ = builder.FieldTypeInt64()
+		case schema.Type_BOOLEAN:
+			typ = builder.FieldTypeBool()
+		case schema.Type_DOUBLE:
+			typ = builder.FieldTypeDouble()
+		case schema.Type_FLOAT:
+			typ = builder.FieldTypeFloat()
+		default:
+			return nil, fmt.Errorf("%s does not have type support in protobuf", p.Type)
```
Comment (toumorokoshi) [inline writer/proto/resource.go:110]: maybe choose an error message that also implies we might just not have that mapping encoded? I don't think we should say that "it doesn't have support" - this isn't exhaustive yet.

Something like "mapping for type %s not found"

**Review by toumorokoshi — APPROVED** [review]
nice! a couple nits, but otherwise LGTM.

File: example/bookstore/v1/bookstore.yaml (hunk 1 above)
Comment (rambleraptor) [reply example/bookstore/v1/bookstore.yaml:20]: Made #21 to discuss.

**Pull request merged**